specification/dartLangSpec.tex

```diff
@@ -9168,33 +9169,52 @@ \subsection{Symbols}
 representing the reserved word \VOID.
 
 \LMHash{}%
-Assume that the term $t$ is an identifier \id{}
-that does not start with an underscore,
-that $t$ is \VOID,
-that $t$ is a period separated sequence of identifiers
-\code{$id_1$.$id_2$.$\cdots$.$id_n$},
-or that $t$ is derived from \synt{operator}.
-Any two occurrences of \code{\#t} evaluate to the same object
-(\commentary{that is, symbol instances are canonicalized}),
-and no other symbol literals evaluate to that object,
-nor to a \code{Symbol} instance that is equal to that object
-according to the \lit{==} operator
+For the value $o$ of a symbol literal representing a source code term
+like an identifier or an operator, we say that $o$ is a
+\IndexCustom{non-private symbol based on}{symbol!non-private, based on}
+the string whose contents is the characters of that term, without whitespace.
+
+\LMHash{}%
+If $o$ is the value of an invocation of the \code{Symbol} constructor
+of the form
+\code{Symbol($e$)}, \code{\NEW\,\,Symbol($e$)},
+or \code{\CONST\,\,Symbol($e$)},
+where $e$ is an expression
+(\commentary{constant if necessary})
+that evaluates to a string $s$,
+we say that $o$ is a \NoIndex{non-private symbol based on} $s$.
+
+\commentary{%
+Note that \code{Symbol('\_foo')} is a non-private symbol,
+and it is distinct from \code{\#\_foo}, as described below.%
+}
+
+\LMHash{}%
+Assume that $i \in 1,2$,
+and that $o_i$ is the value of a constant expression
+which is a symbol based on the string $s_i$.
+If \code{$s_1$\,==\,$s_2$} then $o_1$ and $o_2$ is the same object.
+\commentary{That is, symbol instances are canonicalized.}
+
+\LMHash{}%
+If $o_1$ and $o_2$ are non-private symbols
+(\commentary{not necessarily constant})
+based on strings $s_1$ and $s_2$
+then $o_1$ and $o_2$ are equal according to operator \lit{==}
+if and only if \code{$s_1$ == $s_2$}
 (\ref{equality}).
 
 \LMHash{}%
 A symbol literal \code{\#\_\id} where \id{} is an identifier
```

Comment:
Actually, it should be "where \_\id{} is an identifier" because it covers `#_9` as well.

Reply:
Cf. https://github.com/dart-lang/language/issues/1783 as well.